nexus/db-queries/src/db/datastore/physical_disk.rs

```diff
@@ -126,6 +118,82 @@ impl DataStore {
         Ok(())
     }
 
+    /// Returns all physical disks which:
+    ///
+    /// - Appear on in-service sleds
+    /// - Appear in inventory
+    /// - Do not have any records of expungement
+    pub async fn physical_disk_uninitialized_list(
+        &self,
+        opctx: &OpContext,
+        inventory_collection_id: Uuid,
+        sled_id: Uuid,
+    ) -> ListResultVec<InvPhysicalDisk> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        use db::schema::inv_physical_disk::dsl as inv_physical_disk_dsl;
+        use db::schema::physical_disk::dsl as physical_disk_dsl;
+        use db::schema::sled::dsl as sled_dsl;
+
+        // This is kind of frustrating - we'd like to be able to apply this
+        // condition to all "in service" sleds, regardless of provisionability,
+        // but the sled policy crate explicitly avoids exporting that type.
+        //
+        // As a workaround, we must create the corresponding external types
+        // and manually convert them back into the database types.
```

Comment:
I think this is intentionally difficult because we want to steer queries to using `SledFilter` is to avoid exactly this kind of "build up exactly the right policy / state combination" work. I think we can drop both of these, and change the filter to

```rust
.filter(SledFilter::InService)
```